File: android/app/src/main/kotlin/com/bockmedia/console/local/ClientIdStore.kt
package com.bockmedia.console.local

import android.content.Context
import java.util.UUID

/** Stable install identity for per-device analytics. */
object ClientIdStore {
    private const val PREFS = "bock_client"
    private const val KEY = "client_id"

    fun clientId(context: Context): String {
        val prefs = context.applicationContext.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
        prefs.getString(KEY, null)?.takeIf { it.isNotBlank() }?.let { return it }
        val id = UUID.randomUUID().toString().lowercase()
        prefs.edit().putString(KEY, id).apply()
        return id
    }
}

File: ios/BockMedia/Data/Local/ClientIdStore.swift
import Foundation

/// Stable install identity — stub for Phase 6 per-device accounts.
enum ClientIdStore {
    private static let key = "client_id"

    static func clientId() -> String {
        if let existing = KeychainStore.read(key), !existing.isEmpty {
            return existing
        }
        let id = UUID().uuidString.lowercased()
        KeychainStore.write(key, value: id)
        return id
    }
}
